File: Drive.TypeScript/html/componente/MenuDrive.ts
/// <reference path="../../../Web.TypeScript/erro/Erro.ts"/>
/// <reference path="../../../Web.TypeScript/html/componente/mobile/MenuMobileBase.ts"/>
/// <reference path="../../../Web.TypeScript/html/componente/mobile/MenuMobileItem.ts"/>
/// <reference path="../../../Web.TypeScript/html/componente/mobile/OnMenuClickListener.ts"/>

module Drive
{
    // #region Importações

    import Erro = Web.Erro;
    import MenuMobileBase = Web.MenuMobileBase;
    import MenuMobileItem = Web.MenuMobileItem;
    import OnMenuClickListener = Web.OnMenuClickListener;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export class MenuDrive extends MenuMobileBase implements OnMenuClickListener
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos
        // #endregion Atributos

        // #region Construtores

        constructor()
        {
            super(MenuDrive.name);
        }

        // #endregion Construtores

        // #region Métodos

        protected inicializarItem(arrMmi: Array<MenuMobileItem>): void
        {
        }

        protected setEventos(): void
        {
            super.setEventos();

            PagPrincipalDrive.i.divActionBarDrive.addEvtOnMenuClickListener(this);
        }

        // #endregion Métodos

        // #region Eventos

        public onMenuClick(objSender: Object, arg: JQueryEventObject): void
        {
            try
            {
                this.abrir();
            }
            catch (ex)
            {
                new Erro("Erro desconhecido.", ex);
            }
        }

        // #endregion Eventos
    }
}

File: Principal/Html/Componente/MenuDrive.cs
using NetZ.Web.Html.Componente.Mobile;
using NetZ.Web.Server.Arquivo.Css;

namespace Drive.Html.Componente
{
    internal class MenuDrive : MenuMobileBase
    {
        #region Constantes

        #endregion Constantes

        #region Atributos

        #endregion Atributos

        #region Construtores

        #endregion Construtores

        #region Métodos

        protected override bool getBooJs()
        {
            return true;
        }

        protected override void inicializar()
        {
            base.inicializar();

            this.strId = this.GetType().Name;
        }

        protected override void setCss(CssArquivo css)
        {
            base.setCss(css);

            this.addCss(css.setZIndex(10));
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}